\section{Supplementary notes}

\subsection{Cloning of TIsigner variants of GFP and Luciferase}
The cloning of TIsigner sequence variants for E. coli expression was performed using the MIDAS Golden Gate cloning system \cite{Van_Dolleweerd2018-pw}. As with other Golden Gate assembly (GGA) systems, MIDAS is a modular, hierarchical DNA assembly system that uses the Type IIS restriction enzymes AarI, BsaI and BsmBI to assemble genes, transcription units and other devices from basic parts, and subsequently enables multiple devices to be assembled together on a single plasmid.
As per MIDAS, basic parts such as promoters, coding sequences and terminators were amplified by PCR or ordered as synthetic polynucleotide sequences from gene synthesis companies. The basic parts are listed in Table \ref{tab:appendix_TIsigner_T1}. Protocols for the GGA reactions are as described in van Dolleweerd et al., 2018 \cite{Van_Dolleweerd2018-pw}.

\subsubsection{N-terminal region of GFP (GFPN)}
Overlapping oligonucleotide primer pairs corresponding to the first ten codons of each of the gfp sequence variants produced by the TIsigner algorithm were ordered from Integrated DNA Technologies (IDT) (see Table \ref{tab:appendix_TIsigner_T2}). Each overlapping pair of primers was annealed together and used as a template for amplification by Q5 polymerase (New England Biolabs) to create a double-stranded DNA product spanning the N-terminal region of GFP (GFPN; codons 1 to 10; see Fig \ref{fig:appendix_TIsigner_S1}), and with MIDAS [CCAT] prefix and [GTTG] suffix nucleotides.

\subsubsection{C-terminal region of GFP (GFPC)}
The C-terminal region of GFP (designated GFPC), spanning codons 11 to 238 of the native gfp of \textit{Aequoria victoria}, was synthesized (GeneArt) with flanking BsmBI recognition sites, and with the [GTTG] MIDAS prefix (compatible with the [GTTG] suffix on the GFPN part) and [GCTT] suffix nucleotides.

\subsubsection{N-terminal region of luciferase (RLucN)}
Overlapping oligonucleotide primer pairs corresponding to the first ten codons of each of the luciferase sequence variants generated by the TIsigner algorithm were ordered from Integrated DNA Technologies (IDT) (see Table \ref{tab:appendix_TIsigner_T3}). Each overlapping pair of primers was annealed together and used as a template for amplification by Q5 polymerase (New England Biolabs) to create a double-stranded DNA product spanning the N-terminal region of luciferase (RLucN; codons 1 to 10), and with a MIDAS [CCAT] prefix and an [AGGA] suffix.

\subsubsection{C-terminal region of luciferase (RLucC)}
The C-terminal region of luciferase (designated RLucC), spanning codons 11 to 311 of the native luciferase of \textit{Renilla reniformis}, was amplified from the full-length, native luciferase sequence (synthesized by GeneArt) using primers that add flanking BsmBI recognition sites, and a MIDAS [AGGA] prefix (compatible with the [AGGA] suffix on the RLucN part) and a [GCTT] suffix.


\subsubsection{MIDAS Level-1 cloning of parts}
PCR products, purified using commercially available column-based protocols (Macherey-Nagel), or parts produced by gene synthesis were cloned into the MIDAS pML1 vector by BsmBI-mediated Golden Gate assembly (BsmBI-GGA). As per the MIDAS design, BsmBI-GGA into the pML1 vector results in elimination of the BsmBI recognition sites and each part becomes flanked by BsaI recognition sites that cleave at the MIDAS prefix and suffix nucleotides.

In the case of the \textit{Aequoria victoria} gfp TIsigner variants, each GFPN part cloned into the pML1 vector becomes flanked by BsaI recognition sites that are cleaved at the [CCAT] prefix and [GTTG] suffix (Fig \ref{fig:appendix_TIsigner_S2}). Cloning of the GFPC part into the pML1 vector results in a GFPC module flanked by BsaI recognition sites that are cleaved at the [GTTG] prefix and at the [GCTT] suffix (Fig \ref{fig:appendix_TIsigner_S3}).

In the case of the \textit{Renilla reniformis} luciferase TIsigner variants, each RLucN part cloned into the pML1 vector becomes flanked by BsaI recognition sites that are cleaved at the [CCAT] prefix and [AGGA] suffix, while the C-terminal fragment, RLucC, becomes flanked by BsaI recognition sites that generate an [AGGA] prefix and a [GCTT] suffix upon cleavage.

All parts cloned into the pML1 vector were verified by sequencing.


\subsubsection{MIDAS Level-2 assembly of devices}
Devices were assembled from the cloned Level-1 modules described above, using BsaI-GGA, into the appropriate pML2 vector. As per the MIDAS design, multiple parts can be assembled together, with the position of each part in the assembled device dictated by the compatibility of the prefix and suffix nucleotides flanking each module:
\begin{itemize}
\item A lacI device was assembled in pML2(+)WR from the single lacI genetic element module.
\item An mScarlet-I device was assembled in pML2(+)BR from nptII promoter, mScarlet-I CDS and lambda t0 transcription terminator modules.
\item Full-length gfp devices for each TIsigner variant were assembled in pML2(+)WF from the following modules: T7lac promoter, GFPN, GFPC and T7 T$\phi$ transcription terminator. Since the prefix of the GFPC module, [GTTG] (see Fig \ref{fig:appendix_TIsigner_S3}), is identical to the suffix of each GFPN module (see Fig \ref{fig:appendix_TIsigner_S2}) this allows the two modules to be genetically fused so that, together with the T7lac promoter and T7 T$\phi$ transcription terminator modules, full-length gfp devices are assembled for each variant. The mScarlet-I and gfp devices were assembled in pML2 vectors of opposite orientation (using the “Reverse” vector pML2(+)BR for mScarlet-I, and the “Forward” vector pML2(+)WF for each gfp device), so that they will be divergently transcribed once assembled into the expression vector (Level-3, see below).
\item In a similar fashion, full-length luciferase devices for each TIsigner variant were assembled in pML2(+)BF from T7lac promoter, RLucN, RLucC and T7 T$\phi$ transcription terminator modules.
\end{itemize}
All cloned devices were verified by restriction mapping and sequencing.

\subsubsection{MIDAS Level-3 assembly (construction of the expression plasmids)}
E. coli gfp expression plasmids were constructed by sequentially loading the lacI, mScarlet-I and gfp devices, using alternating AarI- and BsmBI-GGA reactions, into the MIDAS Level-3 destination plasmid pML3.2, which has the medium copy replication origin from the pET series of vectors in place of the high copy pMB1 replicon of the pML3 destination vector originally described in van Dolleweerd et al, 2018 \cite{Van_Dolleweerd2018-pw}. A representative map of an E. coli expression plasmid containing all three devices is shown in Fig \ref{fig:appendix_TIsigner_S4}.
For luciferase expression, the intermediate plasmid containing the lacI device (described above) was used for assembly of each of the luciferase devices (i.e., no mScarlet-I device was added), and a representative map of an E. coli plasmid for luciferase expression is shown in Fig \ref{fig:appendix_TIsigner_S5}. The lacI and luciferase devices are divergently transcribed from the expression vector.

\begin{figure}[htbp!]
	\center
	\includegraphics[width=1\textwidth]{appendix/TIsigner/Figs/S1.png}
	\caption[Strategy for producing a double stranded DNA corresponding to the first ten codons of each of the TIsigner variants of GFP.]{\textbf{Strategy for producing a double stranded DNA corresponding to the first ten codons of each of the TIsigner variants of GFP.} The strategy employs a pair of primers that overlap at their 3’ ends that, upon annealing together, can be used as a template for Q5 polymerase to generate a double stranded DNA spanning the N-terminal region of GFP (i.e. GFPN). Shown here is the sequence of variant GFPN-001 generated using the overlapping cvd2019-09-20a forward and cvd2019-09-20b reverse primer pair (see Table \ref{tab:appendix_TIsigner_T2}). The resultant double stranded DNA can then be cloned into the MIDAS pML1 vector by digestion with the Type IIS restriction enzyme BsmBI (recognition site CGTCTC(1/5)). The same primer pair strategy was used for producing TIsigner variants of luciferase, albeit with a different [AGGA] suffix. This map was created with SnapGene.
		
	}%the List of Figures because of the *}
	\label{fig:appendix_TIsigner_S1}
\end{figure}


\begin{figure}[htbp!]
	\center
	\includegraphics[width=1\textwidth]{appendix/TIsigner/Figs/S2.png}
	\caption[Structure of GFPN variants cloned into the pML1 vector.]{\textbf{Structure of GFPN variants cloned into the pML1 vector.} Following BsmBI-mediated cloning into the pML1 vector, each GFPN variant becomes flanked by BsaI recognition sites (GGTCTC(1/5)). BsaI cleaves at the CCAT prefix upstream of the GFPN sequence and at the GTTG suffix (downstream of the GFPN module). For ease of depiction, only the sequences immediately surrounding the cloned GFPN fragment are shown (i.e., not the rest of the pML1 vector). The structure of luciferase RLucN variants is identical, except for having a different [AGGA] suffix sequence.
		
	}%the List of Figures because of the *}
	\label{fig:appendix_TIsigner_S2}
\end{figure}



\begin{figure}[htbp!]
	\center
	\includegraphics[width=1\textwidth]{appendix/TIsigner/Figs/S3.png}
	\caption[Structure of the GFPC fragment cloned into the pML1 vector.]{\textbf{Structure of the GFPC fragment cloned into the pML1 vector.} Following BsmBI-mediated cloning into the pML1 vector, GFPC becomes flanked by BsaI recognition sites. BsaI cleaves at the [GTTG] prefix (upstream of the GFPC sequence) and at the downstream [GCTT] suffix. For ease of depiction, only sequences around the 5’ and 3’ ends of the GFPC fragment are shown (left- and right-hand sides, respectively). In the case of luciferase, the prefix sequence is [AGGA].
	}%the List of Figures because of the *}
	\label{fig:appendix_TIsigner_S3}
\end{figure}

\begin{figure}[htbp!]
	\center
	\includegraphics[width=1\textwidth]{appendix/TIsigner/Figs/S4.png}
	\caption[Structure of GFP expression plasmids.]{\textbf{Structure of GFP expression plasmids.} Map view showing the architecture of MIDAS-assembled plasmids used for expression of GFP TIsigner variants. Expression of each of the gfp variants is controlled by the T7lac promoter and oriented such that they are divergently transcribed with respect to the mScarlet-I device, which is driven by the nptII promoter. The devices for lacI, mScarlet-I and gfp were loaded sequentially into plasmid pML3.2, which has the medium copy replication origin from the pET series of vectors (ori-bom-rop) in place of the high copy pMB1 replicon in the pML3 destination vector described in van Dolleweerd et al, 2018 \cite{Van_Dolleweerd2018-pw}, and a selectable marker conferring resistance to spectinomycin.
		
	}%the List of Figures because of the *}
	\label{fig:appendix_TIsigner_S4}
\end{figure}

\begin{figure}[htbp!]
	\center
	\includegraphics[width=1\textwidth]{appendix/TIsigner/Figs/S5.png}
	\caption[Structure of luciferase expression plasmids.]{\textbf{Structure of luciferase expression plasmids.} Map view showing the architecture of MIDAS-assembled plasmids used for expression of luciferase TIsigner variants. Expression of each of the luciferase variants is controlled by the T7lac promoter and oriented such that they are divergently transcribed with respect to the lacI device, which is driven by the lacI promoter. The devices for lacI and luciferase were loaded sequentially into plasmid pML3.2, which has the medium copy replication origin from the pET series of vectors (ori-bom-rop) in place of the high copy pMB1 replicon in the pML3 destination vector described in van Dolleweerd et al, 2018 \cite{Van_Dolleweerd2018-pw}, and a selectable marker conferring resistance to spectinomycin.
		
	}%the List of Figures because of the *}
	\label{fig:appendix_TIsigner_S5}
\end{figure}


\section{Supplementary figures}
\begin{figure}[htbp!]
	\center
	\includegraphics[width=1\textwidth]{appendix/TIsigner/Figs/S6.png}
	\caption[Heatmaps of correlations between opening energy and protein abundance for each of the sub-sequence regions (related to Fig 1).]{\textbf{Heatmaps of correlations between opening energy and protein abundance for each of the sub-sequence regions (related to Fig 1).} Green unfilled triangles indicate the regions before and after scaling (left and right panels, respectively). (A) For E. coli, we used a representative GFP expression dataset from Cambray et al. (2018) \cite{Cambray2018-kn}. The reporter library consists of GFP fused in-frame with a library of 96-nt upstream sequences (n=14,425). (B) For S. cerevisiae, we used a YFP expression dataset from Dvir et al. (2013) \cite{Dvir2013-zt}. The YFP reporter library consists of 2,041 random decameric nucleotides inserted at the upstream of YFP start codon. (C) For M. musculus, we used the GFP expression dataset from Noderer et al. (2014) \cite{Noderer2014-aq}. The GFP reporter library consists of 65,536 random hexameric and dimeric nucleotides inserted at the upstream and downstream of GFP start codon, respectively. $R_s$, Spearman’s rho.
	}%the List of Figures because of the *}
	\label{fig:appendix_TIsigner_S6}
\end{figure}

\begin{SCfigure}[][hbtp!]
	\includegraphics[width=0.5\textwidth]{appendix/TIsigner/Figs/S7.png}
	\caption[ Expression outcomes of the PSI:Biology targets in E. coli (related to Fig 2C and 3). ]{\textbf{ Expression outcomes of the PSI:Biology targets in \textit{E. coli} (related to Fig \ref{fig:tisigner_fig2}C and \ref{fig:tisigner_fig3}). } A total of 11,430 PSI:Biology targets from over 189 species were analysed in this study (n=8,780 and 2,650, ‘success’ and ‘failure’ groups, respectively) \cite{Chen2004-is,Seiler2014-ox,Acton2005-sr}. Genera with at least 20 target genes are shown and the remaining as ‘Others’. The top three PSI:Biology targets are from four Pseudomonas, five Bacillus and six Clostridium species. Red asterisk, obelisk and diesis indicate Homo sapiens, S. cerevisiae and E. coli, respectively. These target genes were inserted into the pET21\_NESG expression vector, in which the promoter and fusion tag are T7lac and C-terminal His tag, respectively.}
	\label{fig:appendix_TIsigner_S7}
\end{SCfigure}



\begin{figure}[htbp!]
	\center
	\includegraphics[width=1\textwidth]{appendix/TIsigner/Figs/S8.png}
	\caption[Ribosome footprints in 25-nt fragments show a strong triplet periodicity, indicating translation (related to Fig \ref{fig:tisigner_fig3})]{\textbf{Ribosome footprints in 25-nt fragments show a strong triplet periodicity, indicating translation (related to Fig \ref{fig:tisigner_fig3})}  These 25-nt footprints (green unfilled rectangle) were used to train a neural network model \cite{Tunney2018-kk} in order to predict the translation elongation rates of the PSI:Biology targets. Ribosome profiling data [SRR7759806 and SRR7759807 \cite{Mohammad2019-if}] were first aligned to S. cerevisiae transcriptome. SAM alignment files were merged, and ribosome footprints which were mapped to each frame were enumerated. See https://github.com/Gardner-BinfLab/TIsigner\_paper\_2019. FP, footprints. 
	}%the List of Figures because of the *}
	\label{fig:appendix_TIsigner_S8}
\end{figure}

\begin{figure}[htbp!]
	\center
	\includegraphics[width=1\textwidth]{appendix/TIsigner/Figs/S9.png}
	\caption[Analysis of the local G+C contents in the PSI:Biology target genes (related to Fig \ref{fig:tisigner_fig3}).]{\textbf{Analysis of the local G+C contents in the PSI:Biology target genes (related to Fig \ref{fig:tisigner_fig3}). (A) }  The G+C contents in the regions −24:24 and −30:30 weakly correlate with opening energy and minimum free energy, respectively. Green unfilled squares indicate Spearman’s correlations ($R_s$) between the local G+C contents and the corresponding local features. \textbf{B}  The local G+C contents show a similar prediction accuracy (AUC scores shown in parentheses).
	}%the List of Figures because of the *}
	\label{fig:appendix_TIsigner_S9}
\end{figure}



\begin{SCfigure}[][hbtp!]
	\includegraphics[width=0.5\textwidth]{appendix/TIsigner/Figs/S10.png}
	\caption[Accessibility of translation initiation sites can be increased by synonymous codon substitution within the first nine codons using simulated annealing.]{\textbf{Accessibility of translation initiation sites can be increased by synonymous codon substitution within the first nine codons using simulated annealing. (A)} Schedules in simulated annealing. The ratio of temperature to the number of the first N substitutable codons decreases exponentially with increasing number of iterations. \textbf{(B)} Accessibility of translation initiation sites increases with increasing number of the first N replaceable codons. The PSI:Biology targets that failed to be expressed were optimised using simulated annealing (n=2,650). The Kolmogorov-Smirnov distance between the distributions of ‘9’ and ‘full-length’ was significantly different but sufficiently close (DKS=0.09, P<10-7), indicating that optimisation of the first nine codons can achieve nearly optimum accessibility. For comparison, the distribution of the PSI:Biology targets that were successfully expressed are shown (n=8,780). See also Table S4. \textbf{(C)} Accessibility of translation initiation sites can be increased indirectly using the existing gene optimisation tools and random synonymous codon substitution. ‘TIsigner (9)’ refers to the default settings of our tool, which allows synonymous substitutions up to the first nine codons (as above). See also Table S4. \textbf{(D)} Accessibility of translation initiation sites can be optimised using PCR cloning. The forward primer should be designed according to TIsiger optimised sequences. For example, using a nested PCR approach, the optimised sequence can be produced using the forward primer designed with appropriate mismatches (gold bulges) to amplify the amplicon from the initial PCR reaction.
	}
	\label{fig:appendix_TIsigner_S10}
\end{SCfigure}


\begin{figure}[htbp!]
	\center
	\includegraphics[width=0.5\textwidth]{appendix/TIsigner/Figs/S11.png}
	\caption[Sequence length does not affect software performance because only a fixed region is taken into account during optimisation ($\mathcal{O}(1)$ time). ]{\protect\rule{0ex}{5ex}\textbf{Sequence length does not affect software performance because only a fixed region is taken into account during optimisation ($\mathcal{O}(1)$ time). } 
	}
	\label{fig:appendix_TIsigner_S11}
\end{figure}

\begin{SCfigure}[][hbtp!]
	\includegraphics[width=0.5\textwidth]{appendix/TIsigner/Figs/S12.png}
	\caption[Opening energy of 10 or below at the region −24:24 is about two times more likely to come from the target genes that are successfully expressed than those that failed (related to Fig \ref{fig:tisigner_fig3}). ]{\textbf{Opening energy of 10 or below at the region −24:24 is about two times more likely to come from the target genes that are successfully expressed than those that failed (related to Fig \ref{fig:tisigner_fig3}).} Cumulative frequency distributions of the true positive and false positive (less than type), and true negative and false negative (more than type) derived from the ROC analysis in Fig \ref{fig:tisigner_fig3}B (left panel, opening energy −24:24). These values were used to estimate positive likelihood ratios with 95\% confidence intervals using 10,000 bootstrap replicates. The estimated ratios and/or confidence intervals are inaccurate at low numbers of true positives or true negatives. Therefore, a four-parameter logistic curve was fitted to the positive likelihood ratios. Fitted values are useful to estimate the posterior probability of protein expression.

	}
	\label{fig:appendix_TIsigner_S12}
\end{SCfigure}

\begin{figure}[htbp!]
	\center
	\includegraphics[width=1\textwidth]{appendix/TIsigner/Figs/S13.png}
	\caption[Luciferase reporter assay.]{\textbf{Luciferase reporter assay. (A)} The expression of RLuc can be improved, despite its poor solubility in E. coli. Opening energies are shown next to labels. The luciferase activities of commercially designed RLuc reporter genes (full-length sequence optimisation) and TIsigner (9.9 kcal/mol) are significantly higher than the wild-type luciferase (Mann-Whitney U tests, P=$9.1\times10^{−3}$). \textbf{(B)} SDS-PAGE gel shows the protein bands of Renilla luciferase (RLuc) in the soluble and insoluble fractions of BL21Star(DE3) lysates. Selected bacterial clones were grown at $25^{\circ}$ C, 200 RPM. The solubilities of wildtype (WT) RLuc and designed variants were compared after 4-hour IPTG induction. The blue and red arrows (~36kDa) indicate that RLuc was poorly soluble. No RLuc protein bands were detected from the uninduced cultures and IPTG-induced negative control (empty vector control that lacks Rluc gene and T7lac promoter).
	}
	\label{fig:appendix_TIsigner_S13}
\end{figure}


\begin{figure}[htbp!]
	\center
	\includegraphics[width=1\textwidth]{appendix/TIsigner/Figs/S14.png}
	\caption[The yields of an antibody fragment and an archaebacterial dioxygenase can be improved by synonymous codon changes within the first six codons.]{\textbf{The yields of an antibody fragment and an archaebacterial dioxygenase can be improved by synonymous codon changes within the first six codons.} A RTS \textit{E. coli} cell-free expression system was previously used to express these recombinant proteins (10). The expression levels are shown in arbitrary units (AU) based on the densitometric analysis of previously published Western blots (Supplementary Table S5). WT, wild-type.
	}
	\label{fig:appendix_TIsigner_S14}
\end{figure}

\clearpage

\section{Supplementary tables}



% Please add the following required packages to your document preamble:
% \usepackage{lscape}
% \usepackage{longtable}
% Note: It may be necessary to compile the document several times to get a multi-page table to line up properly
\begin{landscape}
	\begin{longtable}{|l|l|}
		\caption{MIDAS parts used in this work. The sequence of each part is shown, with BsmBI recognition sites used for cloning into the MIDAS pML1 vector. The T7lac promoter and T7 T$\phi$ transcription terminator (both sequences taken from pET-15b) were ordered as double-stranded DNA gBlocks from IDT. The lacI genetic element (with lacI promoter and lacI coding sequence) was from GeneArt. Other parts were amplified by polymerase chain reaction. Sequences of all parts were confirmed following cloning into the MIDAS pML1 vector.
		}
		\label{tab:appendix_TIsigner_T1}\\
		\hline
		\textbf{Part} &
		\textbf{Sequence} \\ \hline
		\endfirsthead
		%
		\multicolumn{2}{c}%
		{{\bfseries Table \thetable\ continued from previous page}} \\
		\hline
		\textbf{Part} &
		\textbf{Sequence} \\ \hline
		\endhead
		%
		T7lac promoter &
		\begin{tabular}[c]{@{}l@{}}cgatgtacgtctcaCTCGGGAGCGATCCCGCGAAATTAATACGACT\\ CACTATAGGGGAATTGTGAGCGGATAACAATTCCCCTCTAGAA\\ ATAATTTTGTTTAACTTTAAGAAGGAGATATACCATtgAGACag\\ agacgaaaggtc\end{tabular} \\ \hline
		nptII promoter &
		\begin{tabular}[c]{@{}l@{}}cgatgtacgtctcaCTCGGGAGctcgcacgctgccgcaagcactcagggcg\\ caagggctgctaaaggaagcggaacacgtagaaagccagtccgcagaaacgg\\ tgctgaccccggatgaatgtcagctactgggctatctggacaagggaaaacgca\\ agcgcaaagagaaagcaggtagcttgcagtgggcttacatggcgatagctaga\\ ctgggcggtcttatggacagcaagcgaaccggaattgccagctggggcgccct\\ ctggtaaggttgggaagccctgcaaagtaaactggatggctttcttgccgccaag\\ gatctgatggcgcaggggatcaagatctgatcaagagacaggatactagtggag\\ gaagaaaaAATGtgAGACagagacgaaaggtc\end{tabular} \\ \hline
		lacI genetic element &
		\begin{tabular}[c]{@{}l@{}}cgatgtacgtctcaCTCGGGAGCGTCGAGATCCCGGACACCATCG\\ AATGGCGCAAAACCTTTCGCGGTATGGCATGATAGCGCCCGG\\ AAGAGAGTCAATTCAGGGTGGTGAATGTGAAACCAGTAACGT\\ TATACGATGTCGCAGAGTATGCCGGTGTCTCTTATCAGACCGT\\ TTCCCGCGTGGTGAACCAGGCCAGCCACGTTTCTGCGAAAA\\ CGCGGGAAAAAGTGGAAGCGGCGATGGCGGAGCTGAATTAC\\ ATTCCCAACCGCGTGGCACAACAACTGGCGGGCAAACAGTC\\ GTTGCTGATTGGCGTTGCCACCTCCAGTCTGGCCCTGCACG\\ CGCCGTCGCAAATTGTCGCGGCGATTAAATCTCGCGCCGAT\\ CAACTGGGTGCCAGCGTGGTGGTGTCGATGGTAGAACGAAG\\ CGGCGTCGAAGCCTGTAAAGCGGCGGTGCACAATCTTCTCG\\ CGCAACGCGTCAGTGGGCTGATCATTAACTATCCGCTGGATG\\ ACCAGGATGCCATTGCTGTGGAAGCTGCCTGCACTAATGTTC\\ CGGCGTTATTTCTTGATGTCTCTGACCAGACACCCATCAACA\\ GTATTATTTTCTCCCATGAAGACGGTACGCGACTGGGCGTGG\\ AGCATCTGGTCGCATTGGGTCACCAGCAAATCGCGCTGTTAG\\ CGGGCCCATTAAGTTCTGTCTCGGCGCGTCTGCGTCTGGCT\\ GGCTGGCATAAATATCTCACTCGCAATCAAATTCAGCCGATAG\\ CGGAACGGGAAGGCGACTGGAGTGCCATGTCCGGTTTTCAA\\ CAAACCATGCAAATGCTGAATGAGGGCATCGTTCCCACTGCG\\ ATGCTGGTTGCCAACGATCAGATGGCGCTGGGCGCAATGCG\\ CGCCATTACCGAGTCCGGGCTGCGCGTTGGTGCGGATATCTC\\ GGTAGTGGGATACGACGATACCGAAGACAGCTCATGTTATATC\\ CCGCCGTTAACCACCATCAAACAGGATTTTCGCCTGCTGGGG\\ CAAACCAGCGTGGACCGCTTGCTGCAACTCTCTCAGGGCCA\\ GGCGGTGAAGGGCAATCAGCTGTTGCCCGTGTCACTGGTGAA\\ AAGAAAAACCACCCTGGCGCCCAATACGCAAACCGCCTCTCC\\ CCGCGCGTTGGCCGATTCATTAATGCAGCTGGCACGACAGGT\\ TTCCCGACTGGAAAGCGGGCAGTGAGCGCAACGCAATTAATG\\ TAAGTTAGCTCACTCATTAGGCACCGGGATCTCGACCGATGCC\\ CTTGAGAGCCTTCAACCCAGTCAGCTCCTTCCGGTGGGCGCG\\ GGGCATGACTACGCTtgAGACagagacgaaaggtc\end{tabular} \\ \hline
		mScarlet-I coding sequence (CDS) &
		\begin{tabular}[c]{@{}l@{}}cgatgtacgtctcaCTCGAATGgtgagcaagggcgaggcagtgatcaaggag\\ ttcatgcggttcaaggtgcacatggagggctccatgaacggccacgagttcgagat\\ cgagggcgagggcgagggccgcccctacgagggcacccagaccgccaagctga\\ aggtgaccaagggtggccccctgcccttctcctgggacatcctgtcccctcagttca\\ tgtacggctccagggccttcatcaagcaccccgccgacatccccgactactataag\\ cagtccttccccgagggcttcaagtgggagcgcgtgatgaacttcgaggacggcg\\ gcgccgtgaccgtgacccaggacacctccctggaggacggcaccctgatctacaa\\ ggtgaagctccgcggcaccaacttccctcctgacggccccgtaatgcagaagaag\\ acaatgggctgggaagcgtccaccgagcggttgtaccccgaggacggcgtgctg\\ aagggcgacattaagatggccctgcgcctgaaggacggcggccgctacctggcg\\ gacttcaagaccacctacaaggccaagaagcccgtgcagatgcccggcgcctaca\\ acgtcgaccgcaagttggacatcacctcccacaacgaggactacaccgtggtgga\\ acagtacgaacgctccgagggccgccactccaccggcggcatggacgagctgta\\ caagtaaGCTTtgAGACagagacgaaaggtc\end{tabular} \\ \hline
		Phage T7 T$\phi$ transcription terminator &
		\begin{tabular}[c]{@{}l@{}}cgatgtacgtctcaCTCGGCTTCAAAGCCCGAAAGGAAGCTGAGTT\\ GGCTGCTGCCACCGCTGAGCAATAACTAGCATAACCCCTTGG\\ GGCCTCTAAACGGGTCTTGAGGGGTTTTTTGCTGAAAGGAGG\\ AACTATATCCGGATCGCTtgAGACagagacgaaaggtc\end{tabular} \\ \hline
		Lambda t0 transcription terminator &
		\begin{tabular}[c]{@{}l@{}}cgatgtacgtctcaCTCGGCTTggactcctgttgatagatccagtaatgacctc\\ agaactccatctggatttgttcagaacgctcggttgccgccgggcgttttttattggt\\ gagaatccaagctagcttggCGCTtgAGACagagacgaaaggtc\end{tabular} \\ \hline
	\end{longtable}
\end{landscape}

%% Please add the following required packages to your document preamble:
%% \usepackage{longtable}
%% Note: It may be necessary to compile the document several times to get a multi-page table to line up properly
%\begin{longtable}{|l|l|}
%	\caption{MIDAS parts used in this work. The sequence of each part is shown, with BsmBI recognition sites used for cloning into the MIDAS pML1 vector underlined. MIDAS prefix and suffix nucleotides are highlighted by the bold blue and red text, respectively. The T7lac promoter and T7 Tφ transcription terminator (both sequences taken from pET-15b) were ordered as double-stranded DNA gBlocks from IDT. The lacI genetic element (with lacI promoter in green and lacI coding sequence in blue) was from GeneArt. Other parts were amplified by polymerase chain reaction. For the mScarlet-I part, the coding sequence is highlighted by the blue shading. Sequences of all parts were confirmed following cloning into the MIDAS pML1 vector.
%	}
%	\label{tab:appendix_TIsigner_T2}\\
%	\hline
%	\textbf{Part} &
%	\textbf{Sequence} \\ \hline
%	\endhead
%	%
%	T7lac promoter &
%	\begin{tabular}[c]{@{}l@{}}cgatgtacgtctcaCTCGGGAGCGATCCCGCGAAATTAATACGACT\\ CACTATAGGGGAATTGTGAGCGGATAACAATTCCCCTCTAGAA\\ ATAATTTTGTTTAACTTTAAGAAGGAGATATACCATtgAGACag\\ agacgaaaggtc\end{tabular} \\ \hline
%	nptII promoter &
%	\begin{tabular}[c]{@{}l@{}}cgatgtacgtctcaCTCGGGAGctcgcacgctgccgcaagcactcagggcg\\ caagggctgctaaaggaagcggaacacgtagaaagccagtccgcagaaacgg\\ tgctgaccccggatgaatgtcagctactgggctatctggacaagggaaaacgca\\ agcgcaaagagaaagcaggtagcttgcagtgggcttacatggcgatagctaga\\ ctgggcggtcttatggacagcaagcgaaccggaattgccagctggggcgccct\\ ctggtaaggttgggaagccctgcaaagtaaactggatggctttcttgccgccaag\\ gatctgatggcgcaggggatcaagatctgatcaagagacaggatactagtggag\\ gaagaaaaAATGtgAGACagagacgaaaggtc\end{tabular} \\ \hline
%	lacI genetic element &
%	\begin{tabular}[c]{@{}l@{}}cgatgtacgtctcaCTCGGGAGCGTCGAGATCCCG\hlc[green]{GACACCATCG}\\ \hlc[green]{AATGGCGCAAAACCTTTCGCGGTATGGCATGATAGCGCCCGG}\\ \hlc[green]{AAGAGAGTCAATTCAGGGTGGTGAAT}\hlc[blue]{GTGAAACCAGTAACGT}\\ \hlc[blue]{TATACGATGTCGCAGAGTATGCCGGTGTCTCTTATCAGACCGT}\\ \hlc[blue]{TTCCCGCGTGGTGAACCAGGCCAGCCACGTTTCTGCGAAAA}\\ \hlc[blue]{CGCGGGAAAAAGTGGAAGCGGCGATGGCGGAGCTGAATTAC}\\ \hlc[blue]{ATTCCCAACCGCGTGGCACAACAACTGGCGGGCAAACAGTC}\\ \hlc[blue]{GTTGCTGATTGGCGTTGCCACCTCCAGTCTGGCCCTGCACG}\\ \hlc[blue]{CGCCGTCGCAAATTGTCGCGGCGATTAAATCTCGCGCCGAT}\\ \hlc[blue]{CAACTGGGTGCCAGCGTGGTGGTGTCGATGGTAGAACGAAG}\\ \hlc[blue]{CGGCGTCGAAGCCTGTAAAGCGGCGGTGCACAATCTTCTCG}\\ \hlc[blue]{CGCAACGCGTCAGTGGGCTGATCATTAACTATCCGCTGGATG}\\ \hlc[blue]{ACCAGGATGCCATTGCTGTGGAAGCTGCCTGCACTAATGTTC}\\ \hlc[blue]{CGGCGTTATTTCTTGATGTCTCTGACCAGACACCCATCAACA}\\ \hlc[blue]{GTATTATTTTCTCCCATGAAGACGGTACGCGACTGGGCGTGG}\\ \hlc[blue]{AGCATCTGGTCGCATTGGGTCACCAGCAAATCGCGCTGTTAG}\\ \hlc[blue]{CGGGCCCATTAAGTTCTGTCTCGGCGCGTCTGCGTCTGGCT}\\ \hlc[blue]{GGCTGGCATAAATATCTCACTCGCAATCAAATTCAGCCGATAG}\\ \hlc[blue]{CGGAACGGGAAGGCGACTGGAGTGCCATGTCCGGTTTTCAA}\\ \hlc[blue]{CAAACCATGCAAATGCTGAATGAGGGCATCGTTCCCACTGCG}\\ \hlc[blue]{ATGCTGGTTGCCAACGATCAGATGGCGCTGGGCGCAATGCG}\\ \hlc[blue]{CGCCATTACCGAGTCCGGGCTGCGCGTTGGTGCGGATATCTC}\\ \hlc[blue]{GGTAGTGGGATACGACGATACCGAAGACAGCTCATGTTATATC}\\ \hlc[blue]{CCGCCGTTAACCACCATCAAACAGGATTTTCGCCTGCTGGGG}\\ \hlc[blue]{CAAACCAGCGTGGACCGCTTGCTGCAACTCTCTCAGGGCCA}\\ \hlc[blue]{GGCGGTGAAGGGCAATCAGCTGTTGCCCGTGTCACTGGTGAA}\\ \hlc[blue]{AAGAAAAACCACCCTGGCGCCCAATACGCAAACCGCCTCTCC}\\ \hlc[blue]{CCGCGCGTTGGCCGATTCATTAATGCAGCTGGCACGACAGGT}\\ \hlc[blue]{TTCCCGACTGGAAAGCGGGCAGTGA}GCGCAACGCAATTAATG\\ TAAGTTAGCTCACTCATTAGGCACCGGGATCTCGACCGATGCC\\ CTTGAGAGCCTTCAACCCAGTCAGCTCCTTCCGGTGGGCGCG\\ GGGCATGACTACGCTtgAGACagagacgaaaggtc\end{tabular} \\ \hline
%	mScarlet-I coding sequence (CDS) &
%	\begin{tabular}[c]{@{}l@{}}cgatgtacgtctcaCTCGA\hlc[blue]{ATGgtgagcaagggcgaggcagtgatcaaggag}\\ \hlc[blue]{ttcatgcggttcaaggtgcacatggagggctccatgaacggccacgagttcgagat}\\ \hlc[blue]{cgagggcgagggcgagggccgcccctacgagggcacccagaccgccaagctga}\\ \hlc[blue]{aggtgaccaagggtggccccctgcccttctcctgggacatcctgtcccctcagttca}\\ \hlc[blue]{tgtacggctccagggccttcatcaagcaccccgccgacatccccgactactataag}\\ \hlc[blue]{cagtccttccccgagggcttcaagtgggagcgcgtgatgaacttcgaggacggcg}\\ \hlc[blue]{gcgccgtgaccgtgacccaggacacctccctggaggacggcaccctgatctacaa}\\ \hlc[blue]{ggtgaagctccgcggcaccaacttccctcctgacggccccgtaatgcagaagaag}\\ \hlc[blue]{acaatgggctgggaagcgtccaccgagcggttgtaccccgaggacggcgtgctg}\\ \hlc[blue]{aagggcgacattaagatggccctgcgcctgaaggacggcggccgctacctggcg}\\ \hlc[blue]{gacttcaagaccacctacaaggccaagaagcccgtgcagatgcccggcgcctaca}\\ \hlc[blue]{acgtcgaccgcaagttggacatcacctcccacaacgaggactacaccgtggtgga}\\ \hlc[blue]{acagtacgaacgctccgagggccgccactccaccggcggcatggacgagctgta}\\ \hlc[blue]{caagtaa}GCTTtgAGACagagacgaaaggtc\end{tabular} \\ \hline
%	Phage T7 T$\phi$ transcription terminator &
%	\begin{tabular}[c]{@{}l@{}}cgatgtacgtctcaCTCGGCTTCAAAGCCCGAAAGGAAGCTGAGTT\\ GGCTGCTGCCACCGCTGAGCAATAACTAGCATAACCCCTTGG\\ GGCCTCTAAACGGGTCTTGAGGGGTTTTTTGCTGAAAGGAGG\\ AACTATATCCGGATCGCTtgAGACagagacgaaaggtc\end{tabular} \\ \hline
%	Lambda t0 transcription terminator &
%	\begin{tabular}[c]{@{}l@{}}cgatgtacgtctcaCTCGGCTTggactcctgttgatagatccagtaatgacctc\\ agaactccatctggatttgttcagaacgctcggttgccgccgggcgttttttattggt\\ gagaatccaagctagcttggCGCTtgAGACagagacgaaaggtc\end{tabular} \\ \hline
%\end{longtable}




\begin{landscape}
	\begin{longtable}{|l|l|l|}
		\caption[Oligonucleotide primer pairs for constructing TIsigner variants of gfp. ]{Oligonucleotide primer pairs for constructing TIsigner variants of gfp. The sequences of each forward and reverse primer pair used for constructing each of the gfp TIsigner variants is shown. The start codon in each of the forward primers is shaded yellow. BsmBI recognition sites (used for Golden Gate assembly into the MIDAS pML1 vector) are underlined.}
%		 The MIDAS prefix [CCAT] and suffix [GTTG] (reverse-complement = CAAC) for the GFPN modules are highlighted in bold blue and red, respectively.}
		\label{tab:appendix_TIsigner_T2}\\
		\hline
		\textbf{\textbf{GFPN TIsigner ID}} & \textbf{Oligonucleotide Primer Pair} & \textbf{Primer Sequences (5’ to 3’)} \\ \hline
		\endfirsthead
		%
		\multicolumn{3}{c}%
		{{\bfseries Table \thetable\ continued from previous page}} \\
		\hline
		\textbf{\textbf{GFPN TIsigner ID}} & \textbf{Oligonucleotide Primer Pair} & \textbf{Primer Sequences (5’ to 3’)} \\ \hline
		\endhead
		%
		\multirow{2}{*}{GFPN-001} & cvd2019-09-20a & \underline{cgatgtacgtctc}aCTCGCCATGAGTAAAGGAGAAGAACTTTTCACTG   \\ \cline{2-3} 
		& cvd2019-09-20b & \underline{gacctttcgtctc}tGTCTcaCAACTCCAGTGAAAAGTTCTTCTCCTTTAC \\ \hline
		\multirow{2}{*}{GFPN-002} & cvd2019-09-21a & \underline{cgatgtacgtctc}aCTCGCCATGTCGAAGGGTGAAGAACTCTTCAC     \\ \cline{2-3} 
		& cvd2019-09-21b & \underline{gacctttcgtctc}tGTCTcaCAACACCAGTGAAGAGTTCTTCACCCTTC  \\ \hline
		\multirow{2}{*}{GFPN-003} & cvd2019-09-21c & \underline{cgatgtacgtctc}aCTCGCCATGAGTAAAGGGGAGGAACTCTTTAC     \\ \cline{2-3} 
		& cvd2019-09-21d & \underline{gacctttcgtctc}tGTCTcaCAACCCCGGTAAAGAGTTCCTCCCCTTTAC \\ \hline
		\multirow{2}{*}{GFPN-004} & cvd2019-09-21e & \underline{cgatgtacgtctc}aCTCGCCATGTCGAAGGGCGAAGAACTCTTC       \\ \cline{2-3} 
		& cvd2019-09-21f & \underline{gacctttcgtctc}tGTCTcaCAACACCAGTGAAGAGTTCTTCGCCCTTC  \\ \hline
		\multirow{2}{*}{GFPN-005} & cvd2019-09-21g & \underline{cgatgtacgtctc}aCTCGCCATGTCTAAGGGTGAGGAGCTCTTC       \\ \cline{2-3} 
		& cvd2019-09-21h & \underline{gacctttcgtctc}tGTCTcaCAACTCCCGTGAAGAGCTCCTCACCCTTAG \\ \hline
		\multirow{2}{*}{GFPN-006} & cvd2019-09-21i & \underline{cgatgtacgtctc}aCTCGCCATGTCGAAAGGGGAAGAACTGTTCAC     \\ \cline{2-3} 
		& cvd2019-09-21j & \underline{gacctttcgtctc}tGTCTcaCAACGCCGGTGAACAGTTCTTCCCCTTTC  \\ \hline
		\multirow{2}{*}{GFPN-007} & cvd2019-09-21k & \underline{cgatgtacgtctc}aCTCGCCATGTCTAAAGGAGAAGAGCTTTTCAC     \\ \cline{2-3} 
		& cvd2019-09-21l & \underline{gacctttcgtctc}tGTCTcaCAACCCCAGTGAAAAGCTCTTCTCCTTTAG \\ \hline
		\multirow{2}{*}{GFPN-008} & cvd2019-09-21m & \underline{cgatgtacgtctc}aCTCGCCATGAGTAAGGGTGAGGAATTATTCACG    \\ \cline{2-3} 
		& cvd2019-09-21n & \underline{gacctttcgtctc}tGTCTcaCAACGCCCGTGAATAATTCCTCACCCTTAC \\ \hline
		\multirow{2}{*}{GFPN-009} & cvd2019-09-22a & \underline{cgatgtacgtctc}aCTCGCCATGAGTAAAGGGGAAGAACTGTTCAC     \\ \cline{2-3} 
		& cvd2019-09-22b & \underline{gacctttcgtctc}tGTCTcaCAACGCCAGTGAACAGTTCTTCCCCTTTAC \\ \hline
		\multirow{2}{*}{GFPN-010} & cvd2019-09-22c & \underline{cgatgtacgtctc}aCTCGCCATGTCTAAGGGTGAGGAGCTCTTC       \\ \cline{2-3} 
		& cvd2019-09-22d & \underline{gacctttcgtctc}tGTCTcaCAACTCCTGTGAAGAGCTCCTCACCCTTAG \\ \hline
		\multirow{2}{*}{GFPN-011} & cvd2019-09-22e & \underline{cgatgtacgtctc}aCTCGCCATGAGTAAAGGAGAAGAGTTATTTACTGG  \\ \cline{2-3} 
		& cvd2019-09-22f & \underline{gacctttcgtctc}tGTCTcaCAACTCCAGTAAATAACTCTTCTCCTTTAC \\ \hline
		\multirow{2}{*}{GFPN-012} & cvd2019-09-22g & \underline{cgatgtacgtctc}aCTCGCCATGAGTAAGGGAGAAGAGCTGTTC       \\ \cline{2-3} 
		& cvd2019-09-22h & \underline{gacctttcgtctc}tGTCTcaCAACTCCAGTGAACAGCTCTTCTCCCTTAC \\ \hline
		\multirow{2}{*}{GFPN-013} & cvd2019-09-22i & \underline{cgatgtacgtctc}aCTCGCCATGTCGAAAGGAGAAGAATTGTTCAC     \\ \cline{2-3} 
		& cvd2019-09-22j & \underline{gacctttcgtctc}tGTCTcaCAACGCCCGTGAACAATTCTTCTCCTTTCG \\ \hline
		\multirow{2}{*}{GFPN-014} & cvd2019-09-22k & \underline{cgatgtacgtctc}aCTCGCCATGAGCAAAGGAGAAGAATTATTTACTGG  \\ \cline{2-3} 
		& cvd2019-09-22l & \underline{gacctttcgtctc}tGTCTcaCAACTCCAGTAAATAATTCTTCTCCTTTGC \\ \hline
		\multirow{2}{*}{GFPN-015} & cvd2019-09-22m & \underline{cgatgtacgtctc}aCTCGCCATGAGCAAAGGAGAAGAATTATTTACGG   \\ \cline{2-3} 
		& cvd2019-09-22n & \underline{gacctttcgtctc}tGTCTcaCAACTCCCGTAAATAATTCTTCTCCTTTGC \\ \hline
		\multirow{2}{*}{GFPN-016} & cvd2019-09-22o & \underline{cgatgtacgtctc}aCTCGCCATGAGCAAAGGGGAAGAATTATTTACAG   \\ \cline{2-3} 
		& cvd2019-09-22p & \underline{gacctttcgtctc}tGTCTcaCAACACCTGTAAATAATTCTTCCCCTTTGC \\ \hline
		\multirow{2}{*}{GFPN-017} & cvd2020-03-05a & \underline{cgatgtacgtctc}aCTCGCCATGAGTAAAGGGGAAGAACTCTTTACC    \\ \cline{2-3} 
		& cvd2020-03-05b & \underline{gacctttcgtctc}tGTCTcaCAACCCCGGTAAAGAGTTCTTCCCCTTTAC \\ \hline
		\multirow{2}{*}{GFPN-018} & cvd2020-03-05c & \underline{cgatgtacgtctc}aCTCGCCATGTCGAAAGGTGAGGAACTATTCACTG   \\ \cline{2-3} 
		& cvd2020-03-05d & \underline{gacctttcgtctc}tGTCTcaCAACACCAGTGAATAGTTCCTCACCTTTC  \\ \hline
		\multirow{2}{*}{GFPN-019} & cvd2020-03-05e & \underline{cgatgtacgtctc}aCTCGCCATGTCGAAGGGTGAAGAACTGTTCACTG   \\ \cline{2-3} 
		& cvd2020-03-05f & \underline{gacctttcgtctc}tGTCTcaCAACACCAGTGAACAGTTCTTCACCCTTC  \\ \hline
		\multirow{2}{*}{GFPN-020} & cvd2020-03-05g & \underline{cgatgtacgtctc}aCTCGCCATGTCGAAGGGTGAAGAACTTTTCACTG   \\ \cline{2-3} 
		& cvd2020-03-05h & \underline{gacctttcgtctc}tGTCTcaCAACCCCAGTGAAAAGTTCTTCACCCTTC  \\ \hline
		\multirow{2}{*}{GFPN-021} & cvd2020-03-05i & \underline{cgatgtacgtctc}aCTCGCCATGTCCAAAGGGGAGGAACTCTTTACG    \\ \cline{2-3} 
		& cvd2020-03-05j & \underline{gacctttcgtctc}tGTCTcaCAACGCCCGTAAAGAGTTCCTCCCCTTTG  \\ \hline
		\multirow{2}{*}{GFPN-022} & cvd2020-03-05k & \underline{cgatgtacgtctc}aCTCGCCATGTCCAAAGGTGAAGAGCTTTTCACC    \\ \cline{2-3} 
		& cvd2020-03-05l & \underline{gacctttcgtctc}tGTCTcaCAACCCCGGTGAAAAGCTCTTCACCTTTG  \\ \hline
		\multirow{2}{*}{GFPN-023} & cvd2020-03-05m & \underline{cgatgtacgtctc}aCTCGCCATGTCGAAAGGTGAAGAGCTGTTCAC     \\ \cline{2-3} 
		& cvd2020-03-05n & \underline{gacctttcgtctc}tGTCTcaCAACACCGGTGAACAGCTCTTCACCTTTC  \\ \hline
		\multirow{2}{*}{GFPN-024} & cvd2020-03-05o & \underline{cgatgtacgtctc}aCTCGCCATGTCGAAAGGTGAGGAACTGTTCAC     \\ \cline{2-3} 
		& cvd2020-03-05p & \underline{gacctttcgtctc}tGTCTcaCAACCCCAGTGAACAGTTCCTCACCTTTC  \\ \hline
		\multirow{2}{*}{GFPN-025} & cvd2020-03-05q & \underline{cgatgtacgtctc}aCTCGCCATGAGTAAGGGGGAGGAGCTCTTCAC     \\ \cline{2-3} 
		& cvd2020-03-05r & \underline{gacctttcgtctc}tGTCTcaCAACTCCGGTGAAGAGCTCCTCCCCCTTAC \\ \hline
		\multirow{2}{*}{GFPN-026} & cvd2020-03-05s & \underline{cgatgtacgtctc}aCTCGCCATGAGTAAAGGGGAAGAGCTTTTCAC     \\ \cline{2-3} 
		& cvd2020-03-05t & \underline{gacctttcgtctc}tGTCTcaCAACCCCGGTGAAAAGCTCTTCCCCTTTAC \\ \hline
		\multirow{2}{*}{GFPN-027} & cvd2020-03-06a & \underline{cgatgtacgtctc}aCTCGCCATGAGTAAAGGAGAAGAACTCTTTACCG   \\ \cline{2-3} 
		& cvd2020-03-06b & \underline{gacctttcgtctc}tGTCTcaCAACTCCGGTAAAGAGTTCTTCTCCTTTAC \\ \hline
		\multirow{2}{*}{GFPN-028} & cvd2020-03-06c & \underline{cgatgtacgtctc}aCTCGCCATGAGTAAAGGAGAAGAACTCTTCACC    \\ \cline{2-3} 
		& cvd2020-03-06d & \underline{gacctttcgtctc}tGTCTcaCAACACCGGTGAAGAGTTCTTCTCCTTTAC \\ \hline
		\multirow{2}{*}{GFPN-029} & cvd2020-03-06e & \underline{cgatgtacgtctc}aCTCGCCATGTCAAAGGGGGAAGAACTGTTCAC     \\ \cline{2-3} 
		& cvd2020-03-06f & \underline{gacctttcgtctc}tGTCTcaCAACGCCTGTGAACAGTTCTTCCCCCTTTG \\ \hline
		\multirow{2}{*}{GFPN-030} & cvd2020-03-06g & \underline{cgatgtacgtctc}aCTCGCCATGTCGAAAGGCGAGGAACTGTTCAC     \\ \cline{2-3} 
		& cvd2020-03-06h & \underline{gacctttcgtctc}tGTCTcaCAACTCCAGTGAACAGTTCCTCGCCTTTC  \\ \hline
		\multirow{2}{*}{GFPN-031} & cvd2020-03-06i & \underline{cgatgtacgtctc}aCTCGCCATGAGCAAGGGTGAAGAGTTATTCACTG   \\ \cline{2-3} 
		& cvd2020-03-06j & \underline{gacctttcgtctc}tGTCTcaCAACTCCAGTGAATAACTCTTCACCCTTG  \\ \hline
		\multirow{2}{*}{GFPN-032} & cvd2020-03-06k & \underline{cgatgtacgtctc}aCTCGCCATGTCTAAAGGTGAAGAACTATTCACAGG  \\ \cline{2-3} 
		& cvd2020-03-06l & \underline{gacctttcgtctc}tGTCTcaCAACCCCTGTGAATAGTTCTTCACCTTTAG \\ \hline
		\multirow{2}{*}{GFPN-033} & cvd2020-03-06m & \underline{cgatgtacgtctc}aCTCGCCATGTCTAAAGGTGAGGAGCTCTTCAC     \\ \cline{2-3} 
		& cvd2020-03-06n & \underline{gacctttcgtctc}tGTCTcaCAACTCCTGTGAAGAGCTCCTCACCTTTAG \\ \hline
		\multirow{2}{*}{GFPN-034} & cvd2020-03-06o & \underline{cgatgtacgtctc}aCTCGCCATGAGTAAGGGAGAGGAACTGTTCAC     \\ \cline{2-3} 
		& cvd2020-03-06p & \underline{gacctttcgtctc}tGTCTcaCAACCCCTGTGAACAGTTCCTCTCCCTTAC \\ \hline
		\multirow{2}{*}{GFPN-035} & cvd2020-03-06q & \underline{cgatgtacgtctc}aCTCGCCATGTCGAAAGGGGAAGAATTGTTCAC     \\ \cline{2-3} 
		& cvd2020-03-06r & \underline{gacctttcgtctc}tGTCTcaCAACTCCAGTGAACAATTCTTCCCCTTTCG \\ \hline
		\multirow{2}{*}{GFPN-036} & cvd2020-03-06s & \underline{cgatgtacgtctc}aCTCGCCATGAGTAAGGGGGAGGAGCTGTTC       \\ \cline{2-3} 
		& cvd2020-03-06t & \underline{gacctttcgtctc}tGTCTcaCAACTCCTGTGAACAGCTCCTCCCCCTTAC \\ \hline
		\multirow{2}{*}{GFPN-037} & cvd2020-03-07a & \underline{cgatgtacgtctc}aCTCGCCATGAGTAAGGGAGAGGAATTGTTCAC     \\ \cline{2-3} 
		& cvd2020-03-07b & \underline{gacctttcgtctc}tGTCTcaCAACACCCGTGAACAATTCCTCTCCCTTAC \\ \hline
		\multirow{2}{*}{GFPN-038} & cvd2020-03-07c & \underline{cgatgtacgtctc}aCTCGCCATGAGTAAGGGAGAGGAACTTTTCAC     \\ \cline{2-3} 
		& cvd2020-03-07d & \underline{gacctttcgtctc}tGTCTcaCAACTCCCGTGAAAAGTTCCTCTCCCTTAC \\ \hline
		\multirow{2}{*}{GFPN-039} & cvd2020-03-07e & \underline{cgatgtacgtctc}aCTCGCCATGAGTAAAGGAGAGGAGCTTTTCACAG   \\ \cline{2-3} 
		& cvd2020-03-07f & \underline{gacctttcgtctc}tGTCTcaCAACTCCTGTGAAAAGCTCCTCTCCTTTAC \\ \hline
		\multirow{2}{*}{GFPN-040} & cvd2020-03-07g & \underline{cgatgtacgtctc}aCTCGCCATGAGCAAAGGAGAAGAGTTATTTACAGG  \\ \cline{2-3} 
		& cvd2020-03-07h & \underline{gacctttcgtctc}tGTCTcaCAACCCCTGTAAATAACTCTTCTCCTTTGC \\ \hline
		\multirow{2}{*}{GFPN-041} & cvd2020-03-07i & \underline{cgatgtacgtctc}aCTCGCCATGAGCAAAGGAGAGGAATTATTTACG    \\ \cline{2-3} 
		& cvd2020-03-07j & \underline{gacctttcgtctc}tGTCTcaCAACGCCCGTAAATAATTCCTCTCCTTTGC \\ \hline
		\multirow{2}{*}{GFPN-042} & cvd2020-05-15a & \underline{cgatgtacgtctc}aCTCGCCATGAGTAAAGGGGAGGAACTCTTTACTG   \\ \cline{2-3} 
		& cvd2020-05-15b & \underline{gacctttcgtctc}tGTCTcaCAACACCAGTAAAGAGTTCCTCCCCTTTAC \\ \hline
		\multirow{2}{*}{GFPN-043} & cvd2020-05-15c & \underline{cgatgtacgtctc}aCTCGCCATGTCGAAAGGTGAAGAACTTTTCACTG   \\ \cline{2-3} 
		& cvd2020-05-15d & \underline{gacctttcgtctc}tGTCTcaCAACACCAGTGAAAAGTTCTTCACCTTTCG \\ \hline
		\multirow{2}{*}{GFPN-044} & cvd2020-05-15e & \underline{cgatgtacgtctc}aCTCGCCATGAGCAAGGGAGAAGAGCTGTTCACTG   \\ \cline{2-3} 
		& cvd2020-05-15f & \underline{gacctttcgtctc}tGTCTcaCAACGCCAGTGAACAGCTCTTCTCCC     \\ \hline
		\multirow{2}{*}{GFPN-045} & cvd2020-05-15g & \underline{cgatgtacgtctc}aCTCGCCATGAGTAAGGGTGAGGAGTTATTCACG    \\ \cline{2-3} 
		& cvd2020-05-15h & \underline{gacctttcgtctc}tGTCTcaCAACGCCCGTGAATAACTCCTCACCCTTAC \\ \hline
		\multirow{2}{*}{GFPN-046} & cvd2020-05-15i & \underline{cgatgtacgtctc}aCTCGCCATGTCTAAAGGAGAAGAACTCTTCACAGG  \\ \cline{2-3} 
		& cvd2020-05-15j & \underline{gacctttcgtctc}tGTCTcaCAACCCCTGTGAAGAGTTCTTCTCCTTTAG \\ \hline
		\multirow{2}{*}{GFPN-047} & cvd2020-05-15k & \underline{cgatgtacgtctc}aCTCGCCATGTCCAAAGGAGAAGAACTATTCACC    \\ \cline{2-3} 
		& cvd2020-05-15l & \underline{gacctttcgtctc}tGTCTcaCAACTCCGGTGAATAGTTCTTCTCCTTTGG \\ \hline
		\multirow{2}{*}{GFPN-048} & cvd2020-05-15m & \underline{cgatgtacgtctc}aCTCGCCATGAGCAAAGGAGAAGAACTATTCACG    \\ \cline{2-3} 
		& cvd2020-05-15n & \underline{gacctttcgtctc}tGTCTcaCAACTCCCGTGAATAGTTCTTCTCCTTTGC \\ \hline
		\multirow{2}{*}{GFPN-049} & cvd2020-05-15o & \underline{cgatgtacgtctc}aCTCGCCATGTCGAAGGGAGAAGAATTATTTACGG   \\ \cline{2-3} 
		& cvd2020-05-15p & \underline{gacctttcgtctc}tGTCTcaCAACTCCCGTAAATAATTCTTCTCCCTTCG \\ \hline
		
		
	\end{longtable}
	
\end{landscape}


% Please add the following required packages to your document preamble:
% \usepackage{multirow}
% \usepackage{graphicx}
\begin{table}[]
	\caption{Oligonucleotide primer pairs for constructing TIsigner variants of luciferase. The sequences of each forward and reverse primer pair used for constructing each of the luciferase TIsigner variants is shown. The start codon in each of the forward primers is shaded yellow. BsmBI recognition sites (used for Golden Gate assembly into the MIDAS pML1 vector) are underlined. }
	\label{tab:appendix_TIsigner_T3}
	\resizebox{\textwidth}{!}{%
		\begin{tabular}{|l|l|l|}
			\hline
			\textbf{RLucN TIsigner ID}    & \textbf{Oligonucleotide Primer Pair} & \textbf{Primer Sequences (5’ to 3’)}                 \\ \hline
			\multirow{2}{*}{RLucN-TI-002} & cvd2019-06-14c & cgatgta\underline{cgtctc}aCTCGCCATGACATCAAAAGTATACGACCCAGAG      \\ \cline{2-3} 
			& cvd2019-06-14d & gaccttt\underline{cgtctc}tGTCTcaTCCTCTGCTCTGGGTCGTATACTTTTGATG   \\ \hline
			\multirow{2}{*}{RLucN-TI-003} & cvd2019-06-14e & cgatgta\underline{cgtctc}aCTCGCCATGACAAGTAAAGTTTATGACCCAGAGC     \\ \cline{2-3} 
			& cvd2019-06-14f & gaccttt\underline{cgtctc}tGTCTcaTCCTCTGCTCTGGGTCATAAACTTTACTTG   \\ \hline
			\multirow{2}{*}{RLucN-TI-004} & cvd2019-06-14g & cgatgta\underline{cgtctc}aCTCGCCATGACCAGCAAAGTTTATGACCCAGAG      \\ \cline{2-3} 
			& cvd2019-06-14h & gaccttt\underline{cgtctc}tGTCTcaTCCTCTGCTCTGGGTCATAAACTTTGCTG    \\ \hline
			\multirow{2}{*}{RLucN-TI-005} & cvd2019-06-14i & cgatgta\underline{cgtctc}aCTCGCCATGACAAGCAAAGTTTATGACCCAGAGC     \\ \cline{2-3} 
			& cvd2019-06-14j & gaccttt\underline{cgtctc}tGTCTcaTCCTCTGCTCTGGGTCATAAACTTTGC      \\ \hline
			\multirow{2}{*}{RLucN-TI-006} & cvd2019-06-14k                       & cgatgta\underline{cgtctc}aCTCGCCATGACTTCGAAAGTTTATGATCCAGAACAG   \\ \cline{2-3} 
			& cvd2019-06-14l & gaccttt\underline{cgtctc}tGTCTcaTCCTCTGTTCTGGATCATAAACTTTCGAAG   \\ \hline
			\multirow{2}{*}{RLucN-TI-007} & cvd2019-06-14m                       & cgatgta\underline{cgtctc}aCTCGCCATGACATCAAAAGTTTATGATCCAGAACAAAG \\ \cline{2-3} 
			& cvd2019-06-14n & gaccttt\underline{cgtctc}tGTCTcaTCCTTTGTTCTGGATCATAAACTTTTGATGTC \\ \hline
			\multirow{2}{*}{RLucN-TI-008} & cvd2019-06-14o & cgatgta\underline{cgtctc}aCTCGCCATGACGTCGAAAGTTTACGATCCAG        \\ \cline{2-3} 
			& cvd2019-06-14p & gaccttt\underline{cgtctc}tGTCTcaTCCTTTGTTCTGGATCGTAAACTTTCGACG   \\ \hline
			\multirow{2}{*}{RLucN-TI-009} & cvd2019-06-14q & cgatgta\underline{cgtctc}aCTCGCCATGACATCGAAAGTTTACGATCCAGAAC     \\ \cline{2-3} 
			& cvd2019-06-14r & gaccttt\underline{cgtctc}tGTCTcaTCCTTTGTTCTGGATCGTAAACTTTCGATG   \\ \hline
			\multirow{2}{*}{RLucN-TI-010} & cvd2019-06-14s & cgatgta\underline{cgtctc}aCTCGCCATGACCTCGAAAGTTTATGACCCAGAAC     \\ \cline{2-3} 
			& cvd2019-06-14t & gaccttt\underline{cgtctc}tGTCTcaTCCTTTGTTCTGGGTCATAAACTTTCGAG    \\ \hline
		\end{tabular}%
	}
\end{table}


